src/main/java/Rabbit/util/TaskList.java

```diff
@@ -0,0 +1,203 @@
+package Rabbit.util;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+import Rabbit.RabbitException.FindFormatException;
+import Rabbit.Task.Task;
+import Rabbit.Task.Event;
+import Rabbit.Task.Deadline;
+import Rabbit.Task.Todo;
+import Rabbit.RabbitException.AddToListException;
+import Rabbit.RabbitException.MarkUnmarkException;
+import Rabbit.RabbitException.DeleteException;
+
+
+public class TaskList {
+    private ArrayList<Task> list;
+    public enum TaskType {
+        TODO, DEADLINE, EVENT;
+    }
+
+    public TaskList() {
+        this.list = new ArrayList<>();
+    }
+
+    /**
+     * Returns tasks in the list that match
+     * the searching keyword.
+     *
+     * @param input the searching keyword.
+     * @return the tasks matching the searching keyword.
+     * @throws FindFormatException
+     */
+    public String find(String input) throws FindFormatException {
+        try {
+            String keyword = Parser.parseFind(input, this);
+            if (keyword.equals(" ")) {
+                throw new FindFormatException();
+            }
+            String output = "";
+            for (int i = 0; i < this.list.size(); i++) {
+                String content = this.list.get(i).getContent();
+
+                if (content.length() < keyword.length()) {
+                    continue;
+                }
+
+                for (int j = 0; j < content.length() - keyword.length() + 1; j++) {
```

Comment:
Perhaps you could consider adding a method in the Task class/subclasses to check if they contain a certain word? That way, it might add to OOP and code quality?